src/t_zset.c

```diff
@@ -1367,7 +1366,10 @@ int zsetAdd(robj *zobj, double score, sds ele, int in_flags, int *out_flags, dou
             {
                 zobj->ptr = zzlDelete(zobj->ptr,eptr);
                 zobj->ptr = zzlInsert(zobj->ptr,ele,score);
+                if (newscore) *newscore = score;
```

Comment:
could be `if (incr && newscore)` in all occurrences

Reply:
i think it's a good idea to keep it as it's now, and always updated `newscore` regardless of the incr option (unless out_flags is ZADD_OUT_NOP)